### DISCUSSION TITLE
How do I create a kinematic body?

### DISCUSSION THREAD

**Discussion by OhJeongrok** [notgiven688/jitterphysics2#68] [How do I ... ?]
To use jitter2 on the server
I want to receive the character's movement from the client.
So, on the server, I want to handle the character's rigid body as kinematic. How can I do this?

**Comment by notgiven688** [comment]
Do you just want to set the position on the server? Then make the body static and move it around by setting the position.

Here is an example for a "traditional" kinematic object:

```
box = world.CreateRigidBody();
box.AddShape(new BoxShape(1));
box.Position = new JVector(0, 3f, 0);
box.SetMassInertia(JMatrix.Zero, 0, true);  // <- infinite mass
box.AffectedByGravity = false;
box.Velocity = new JVector(10, 0, 0);
box.Damping = (0, 0);
```

The box moves with a constant velocity and is unstoppable. Make sure to filter out any collision between this object and any static object, otherwise the solver might explode (unstoppable force vs immovable object).

**Comment by OhJeongrok** [comment]
thank you!

**Comment by Khhs167** [comment]
I am also looking into Jitter Physics for my game. I don't need any complex simulation, but I do need a character controller. I will preferably provide velocity myself, the only real feature I need is for it to not phase through walls, and slide appropriately(so, something likes godot's move_and_slide).

Could this be done with Jitter 2? And if, how?

If not, I strongly recommend you add it to the engine, since kinematic bodies are really useful.

**Reply by notgiven688** [reply]
Hi @Khhs167,

in the "level geometry"-demo scene there is a small capsule-character which you can move around using the arrow-keys. The basic character controller is implemented in Player.cs (https://github.com/notgiven688/jitterphysics2/blob/main/src/JitterDemo/Demos/Player/Player.cs). 

Does this help? Feel free to ask further questions. I should add some info in the documentation.

You can also just set the velocity directly every frame. Jitter will detect any collision and bring the character to a halt before integrating the velocities to get new positions - the body will never enter the wall.

**Reply by long1752062104** [reply]
> 你好@Khhs167，
> 
> 在“关卡几何”演示中，有一个小胶囊角色，您可以使用箭头键移动它。基本角色控制器场景在 Player.cs 中实现（[https://github.com/notgiven688/jitterphysicals2/blob/main/ src/JitterDemo/Demos/Player/Player.cs](https://github.com/notgiven688/jitterphysics2/blob/main/src/JitterDemo/Demos/Player/Player.cs)）。
> 
> 这有帮助吗？请随意提问。我应该在文档中添加一些信息。
> 
> 你也可以直接设置每一帧的速度。

I found that this character got stuck near the wall during the testing process and couldn't move backwards or slide properly
